src/braket/circuits/gate.py

```diff
@@ -210,6 +211,32 @@ def __eq__(self, other):
     def __repr__(self):
         return f"{self.name}('qubit_count': {self._qubit_count})"
 
+    def __hash__(self):
+        return hash((self.name, self.qubit_count))
+
+    @staticmethod
+    def _str_to_gate(class_name: str) -> Gate:
+        """
+        Returns the class of Gate corresponding to the string assigned.
```

Comment:
- Which string assigned? I don't understand where these strings are relevant. 
- Why do you need an entire method with if/else statements? Could this translation be performed using a Dict?

Reply:
Fixed the docstring

Changed to a dict for this class and the parsing within Waveforms.